KEXPPower/Helpers/ListenerId.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  ListenerId.swift
+//  KEXPPower
+//
+//  Created by Adam Bourn on 1/22/22.
+//  Copyright © 2022 KEXP. All rights reserved.
+//
+
+import Foundation
+
+
+class ListenerId {
+    // Singleton ListenerId
+    public static let sharedInstance = ListenerId()
```

Comment:
Does this need to be public? Does this need to be a singleton?

Reply:
`KEXPPower` is already a singleton, could potentially init/store the `listenerId` in that object
`append(toURL url: URL)` becomes a static utility function and would need to pass in `listenerId` - `ArchiveManger` could grab that id from `KEXPPower`
This avoids a singleton for `ListenerId`, @dbergman is this your general thinking?

Reply:
@jeffsorr exactly, don't think `ListenerId` class is needed.   Ideally the listenID parameter would be appended in the NetworkManager or ArchiveManager...

 I going to address a few items/refactor KEXPPower in a bit...may find a  better place for it.

Reply:
My thinking with having a dedicated class here was related to the fact that a `KEXPPower` instance can be instantiated with specific stream URLs (which will need to have `listenerId` appended  *and* the fact that we fetch archive stream URLs via the separate `ArchiveManager`.  I agree, I would have preferred this logic to live in either the `NetworkManager` or `ArchiveManager` instead, if it was the case that `KEXPPower` couldn't be initialized with particular stream URLs.

Basically, I was trying to avoid having the `append` logic in the `KEXPPower` class so that the `ArchiveManager` wouldn't have to make a call to a static utility function in `KEXPPower`.  Idea was to move it out to a separate utility entirely, since access/functionality is needed in multiple places.  Making `ListenerId` a singleton was just to add an extra protection against a new instance being created and consequently a new `listenerId` being generated for a given user.

That said, it sounds like the consensus is that the dedicated class is overkill, so with that in mind I'll just move this to `KEXPPower` and have the `ArchiveManager` call into that. Since KEXPPower is already a singleton though, I'll probably keep `append` as an instance method, rather than a static function (unless @jeffsorr 
@dbergman you were suggesting a global function).

Reply:
@jeffsorr @dbergman Just pushed the commits to implement this ^^^, please take a look when you have a moment. Thanks for the review!

Reply:
@abourn This looks good to me, I have working branch bringing keeping the streamURLs internal to this dependency.  

So some further adjustments may show up in a PR when its ready